EFCore.NamingConventions/Internal/NamingConventionsOptionsExtension.cs

```diff
@@ -74,6 +77,14 @@ public virtual NamingConventionsOptionsExtension WithCamelCaseNamingConvention(C
         return clone;
     }
 
+    public virtual NamingConventionsOptionsExtension WithCustomNamingConvention(Func<string, string> mapping)
+    {
+        var clone = Clone();
+        clone._namingConvention = NamingConvention.Custom;
+        clone._customMapper = mapping;
```

Comment:
We should reset _customMapper to null in the other With* methods to keep things clean